core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -64,22 +66,25 @@ public static TableMetadata newTableMetadata(TableOperations ops,
                                                PartitionSpec spec,
                                                String location,
                                                Map<String, String> properties) {
-    return newTableMetadata(schema, spec, SortOrder.unsorted(), location, properties, DEFAULT_TABLE_FORMAT_VERSION);
+    int formatVersion = PropertyUtil.propertyAsInt(properties, FORMAT_VERSION, FORMAT_VERSION_DEFAULT);
+    return newTableMetadata(schema, spec, SortOrder.unsorted(), location, properties, formatVersion);
```

Comment:
Right now, we don't make it easy to create a v2 table because we don't want people actually using them. That's why all of the tests currently create a table and then update it to v2.

We would also need to remove `FORMAT_VERSION` from properties because we don't intend to keep a table property in sync with the table version in metadata.

Reply:
OK,  I can change to use the `TestTables`  to test the format v2.  If we think all of the v2 work are finished, then we open to expose it to users.

Reply:
Yes, that's the plan.